lib/matplotlib/backend_tools.py

```diff
@@ -452,22 +453,18 @@ def add_figure(self):
         if self.figure not in self.views:
             self.views[self.figure] = cbook.Stack()
             self.positions[self.figure] = cbook.Stack()
+            self.home_views[self.figure] = WeakKeyDictionary()
             # Define Home
             self.push_current()
-            # Adding the clear method as axobserver, removes this burden from
-            # the backend
-            self.figure.add_axobserver(self.clear)
-
-    def clear(self, figure):
```

Comment:
I just added `clear` back in with
```py
self.home_views[figure].clear()
self.update_home_views()
```
added.  That's the only change in this new commit.